gapis/api/gles/dependency_graph_behaviour_provider.go

```diff
@@ -182,34 +186,34 @@ func (*GlesDependencyGraphBehaviourProvider) GetBehaviourForAtom(
 				if cmd.SrcTarget == GLenum_GL_RENDERBUFFER {
 					b.Read(g, renderbufferDataKey{c.Objects.Shared.Renderbuffers[RenderbufferId(cmd.SrcName)]})
 				} else {
-					data, size := c.Objects.Shared.Textures[TextureId(cmd.SrcName)].dataAndSize()
+					data, size := c.Objects.Shared.Textures[TextureId(cmd.SrcName)].dataAndSize(cmd.SrcLevel, 0)
 					b.Read(g, data)
 					b.Read(g, size)
 				}
 				if cmd.DstTarget == GLenum_GL_RENDERBUFFER {
 					b.Write(g,
 						renderbufferDataKey{c.Objects.Shared.Renderbuffers[RenderbufferId(cmd.DstName)]})
 				} else {
-					data, size := c.Objects.Shared.Textures[TextureId(cmd.DstName)].dataAndSize()
+					data, size := c.Objects.Shared.Textures[TextureId(cmd.DstName)].dataAndSize(cmd.DstLevel, 0)
 					b.Write(g, data)
 					b.Write(g, size)
 				}
 			case *GlFramebufferTexture2D:
-				b.Read(g, textureSizeKey{c.Objects.Shared.Textures[cmd.Texture], cmd.Texture})
+				b.Read(g, textureSizeKey{c.Objects.Shared.Textures[cmd.Texture], cmd.Texture, cmd.Level, 0})
 				b.KeepAlive = true // Changes untracked state
 			case *GlCompressedTexImage2D:
-				texData, texSize := getTextureDataAndSize(ctx, cmd, id, s, c.Bound.TextureUnit, cmd.Target)
+				texData, texSize := getTextureDataAndSize(ctx, cmd, id, s, c.Bound.TextureUnit, cmd.Target, cmd.Level, 0)
 				b.Modify(g, texData)
 				b.Write(g, texSize)
 			case *GlCompressedTexSubImage2D:
-				texData, _ := getTextureDataAndSize(ctx, cmd, id, s, c.Bound.TextureUnit, cmd.Target)
+				texData, _ := getTextureDataAndSize(ctx, cmd, id, s, c.Bound.TextureUnit, cmd.Target, cmd.Level, 0)
 				b.Modify(g, texData)
 			case *GlTexImage2D:
-				texData, texSize := getTextureDataAndSize(ctx, cmd, id, s, c.Bound.TextureUnit, cmd.Target)
+				texData, texSize := getTextureDataAndSize(ctx, cmd, id, s, c.Bound.TextureUnit, cmd.Target, cmd.Level, 0)
 				b.Modify(g, texData)
 				b.Write(g, texSize)
 			case *GlTexSubImage2D:
-				texData, _ := getTextureDataAndSize(ctx, cmd, id, s, c.Bound.TextureUnit, cmd.Target)
+				texData, _ := getTextureDataAndSize(ctx, cmd, id, s, c.Bound.TextureUnit, cmd.Target, cmd.Level, 0)
```

Comment:
All of the code above sets layer to 0, which is wrong.
It should be CubemapFaceToLayer(cmt.Target).
As it is
  TexImage2D(POSITVE_X, ....)
  TexImage2D(NEGATIVE_X, ....)
will keep only the last face upload alive.

Reply:
Fixed